src/components/header/AppHeaderSearch.vue

```diff
@@ -0,0 +1,132 @@
+<script setup lang="ts">
+import { useQuasar } from 'quasar';
+import { ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { useRouter } from 'vue-router';
+
+import { evm } from 'src/boot/evm';
+
+
+const $router = useRouter();
+const $q = useQuasar();
+const { t: $t } = useI18n();
+
+const TIME_DELAY = 6000;
+
+const searchTerm = ref('');
+const inputRef = ref<HTMLInputElement | null>(null);
+
+async function search() {
+    if (!searchTerm.value) {
+        return;
+    }
+
+    searchTerm.value = searchTerm.value.trim().replace(/\s/, '');
+    if (searchTerm.value.startsWith('0x')) {
+        if (searchTerm.value.length === 42) {
+            $router.push(`/address/${searchTerm.value}`);
+            return;
+        } else {
+            $router.push(`/tx/${searchTerm.value}`);
+            return;
+        }
+    } else if (searchTerm.value.match(/(^[a-z1-5.]{1,11}[a-z1-5]$)|(^[a-z1-5.]{12}[a-j1-5]$)/)) {
```

Comment:
This RegEx is complex enough to deserve its own commented explanation

(That's another way to say I didn't get a thing of what it says there lmao 🤣 )

Reply:
Yeah that was there when I joined telos haha. I believe it checks if the string is a valid telos native account name... maybe I can verify that and leave a comment